src/core/SampleCache.cpp

```diff
@@ -32,22 +32,43 @@
 namespace lmms {
 auto SampleCache::fetch(const QString& path) -> std::shared_ptr<SampleBuffer>
 {
-	// To ensure that only the main thread is requesting samples from the cache
-	// Can be removed if proven that other threads are requesting samples besides the main thread
-	assert(QThread::currentThread() == QCoreApplication::instance()->thread());
+	namespace fs = std::filesystem;
 
-	const auto fsPath = PathUtil::pathFromQString(PathUtil::toAbsolute(path));
-	auto entry = AudioFileEntry{fsPath, std::filesystem::last_write_time(fsPath)};
-	return get(std::move(entry), s_audioFileMap, path);
+	const auto it = std::find_if(s_audioFileMap.begin(), s_audioFileMap.end(),
+		[&](const auto& entry) { return entry.first.path == PathUtil::pathFromQString(path); });
+
+	auto lastWriteTime = fs::last_write_time(PathUtil::pathFromQString(path));
+
+	if (it == s_audioFileMap.end() || it->first.lastWriteTime != lastWriteTime)
+	{
+		const auto buffer = std::make_shared<SampleBuffer>(path);
+		const auto key = AudioFileEntry{PathUtil::pathFromQString(path), lastWriteTime};
+		s_audioFileMap[std::move(key)] = buffer;
+		return buffer;
+	}
```

Comment:
```suggestion
	const auto qPath = PathUtil::pathFromQString(path);
	const auto it = std::find_if(s_audioFileMap.begin(), s_audioFileMap.end(),
		[&](const auto& entry) { return entry.first.path == qPath; });

	auto lastWriteTime = fs::last_write_time(qPath);

	if (it == s_audioFileMap.end() || it->first.lastWriteTime != lastWriteTime)
	{
		const auto buffer = std::make_shared<SampleBuffer>(path);
		const auto key = AudioFileEntry{qPath, lastWriteTime};
		s_audioFileMap[std::move(key)] = buffer;
		return buffer;
	}
```
Moved the `PathUtil::pathFromQString(path)` outside the lambda so it doesn't call it every time the lambda is called.

Reply:
I would use `fsPath` instead personally, but yeah I'll move it out.